worker/src/main.rs

```diff
@@ -188,31 +197,88 @@ fn worker(node_url: &str, w_ip: &str, w_port: &str, mu_ra_port: &str) {
 
 	// ------------------------------------------------------------------------
 	// start the substrate-api-client to communicate with the node
-	let mut api = Api::new(format!("ws://{}", node_url));
-	api.init();
+	let alice = primitives::sr25519::Pair::from_string("//Alice", None).unwrap();
+	println!("   Alice's account = {}", alice.public().to_ss58check());
+	let alicekey = AccountKey::Sr(alice.clone());
+	// alice is validator
+	let mut api = Api::new(format!("ws://{}", node_url))
+	   	.set_signer(alicekey.clone());
+
+	info!("encoding Alice public 	= {:?}", alice.public().0.encode());
+	info!("encoding Alice AccountId = {:?}", AccountId::from(alice.public()).encode());
+
+	let result_str = api.get_storage("Balances", "FreeBalance", Some(AccountId::from(alice.public()).encode())).unwrap();
+    let funds = hexstr_to_u256(result_str).unwrap();
+	println!("    Alice's free balance = {:?}", funds);
+    let result_str = api.get_storage("System", "AccountNonce", Some(AccountId::from(alice.public()).encode())).unwrap();
+    let result = hexstr_to_u256(result_str).unwrap();
+    println!("    Alice's Account Nonce is {}", result.low_u32());
+
 
 	// ------------------------------------------------------------------------
 	// get required fields for the extrinsic
-	let genesis_hash = api.genesis_hash.unwrap().as_bytes().to_vec();
+	let genesis_hash = api.genesis_hash.as_bytes().to_vec();
 
 	// get the public signing key of the TEE
-	let mut key = [0; 32];
-	let ecc_key = fs::read(ECC_PUB_KEY).expect("Unable to open ECC public key file");
-	key.copy_from_slice(&ecc_key[..]);
-	info!("[+] Got ECC public key of TEE = {:?}", key);
+	println!("*** Ask the signing key from the TEE");
+	let tee_pubkey_size = 32;
+	let mut tee_pubkey = [0u8; 32];
```

Comment:
why not use tee_pubkey_size directly here?

Reply:
possible, but cosmetic